test/m4/miscellaneous.m4

```diff
@@ -4,5 +4,11 @@
 AC_ARG_ENABLE([test-cpp],
     [AS_HELP_STRING([--disable-test-cpp], [run basic test aginast cpp])],
     [test_cpp=$enableval],
-    [test_cpp=yes])
+    [AC_LANG_PUSH([C++])
```

Comment:
Missing AC_PROG_CXX.

Also I think it's better to write it in the following way:
```autoconf
AC_ARG_ENABLE([.......],
    [test_cpp=$enableval],
    [test_cpp=check])

if test "x$test_cpp" != "xno" ; then
    AC_PROG_CXX
    AC_CACHE_CHECK([if C++ compiler works], [odp_cv_cxx_works],
      [AC_COMPILE_IFELSE([....], [odp_cv_cxx_works=yes], [odp_cv_cxx_works=no])])
   # or you can rewrite this as two inline diffs
    AS_IF([test "x$test_cpp$odp_cv_cxx_works" = "xyesno"], [AC_MSG_FAILURE([...])],
               [test "x$test_cpp$odp_cv_cxx_works" = "xcheckno"], [AC_MSG_NOTICE([disabling C++ test])])
fi

```

Reply:
> Missing AC_PROG_CXX.

AC_PROG_CXX is already called in configure.ac. Should it be called here also? My autoconf understanding is rather limited =/

>if test "x$test_cpp" != "xno" ; then
    AC_PROG_CXX
    AC_CACHE_CHECK([if C++ compiler works], [odp_cv_cxx_works],
      [AC_COMPILE_IFELSE([....], [odp_cv_cxx_works=yes], [odp_cv_cxx_works=no])])
...

This doesn't seem to work (test passes even though there is no C++ compiler). If I replace AC_PROG_CXX with AC_LANG_PUSH([C++]) and AC_LANG_POP([C++]) it works.
```
    AC_LANG_PUSH([C++])
    AC_CACHE_CHECK([if C++ compiler works], [odp_cv_cxx_works],
      [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([])], [odp_cv_cxx_works=yes], [odp_cv_cxx_works=no])])
    AC_LANG_POP([C++])
```

Reply:
Maybe I missed a call for `AC_PROG_CXX`. You can skip it then.
Please put `AC_LANG_PUSH`/`AC_LANG_POP` inside AC_CACHE_CHECK argument. Then I will be fine with this version.

Reply:
Will do, thanks!